Expensify/App
# QBO authentication

This is just a quick prototype to test the QBO authentication flow and disconnection flow

Needs:

- https://github.com/Expensify/Web-Expensify/pull/40741
- https://github.com/Expensify/Integration-Server/pull/7804


### Details


### Fixed Issues

$
PROPOSAL:


### Tests

1. Create a free workspace
2. Go to the workspace settings
3. Click the new menu entry `Accounting`
4. Click "Setup" in the QBO menu entry
5. Verify
    - In Web it opens a new tab with the QBO OAuth page
    - In Native (IOS/Android) it opens a WebView with the QBO OAuth page
7. Use the `qa@expensify.com` account in 1Password to authenticate
8. Verify that after authenticating the tab or the WebView are closed
9. Verify that a spinner appears in the right side of the QBO menu entry and the supporting text changes with the import step
10. After the import is finished, click the 3 vertical dots menu
11. Click "Disconnect"
12. Verify that the QBO menu entry returned to the start: the button "Setup" is there again


https://github.com/Expensify/Integration-Server/assets/87341702/0c69d177-fadc-41a6-98c3-bb32d4c906bc

https://github.com/Expensify/Integration-Server/assets/87341702/f363a6e5-ae5c-44d8-8796-bc3e2d553d1f
